### PR TITLE
AVSR recipe for Easycom Dataset

### PR DESCRIPTION
**Pull request by ms-dot-k**
## What?

New recipe for training audio-visual speech recognition model on Easycom dataset.
The recipe is based on LRS3 avsr recipe which utilizes pre-trained AV-HuBERT model. (Dumped features)

I added data augmentation techniques to the espnet2/asr/encoder/avhubert_encoder.py
1) acoustic noise perturbation: a babble noise is corrupted with random noisy strengths at the feature level.
2) modality dropout: audio and video streams are randomly dropped out, so that we can still perform audio-visual or audio-only, visual-only prediction after the model is trained.

## See also

Easycom dataset is too small to achieve proper performances by using the dataset only. The recipe utilizes both Easycom and LRS3 datasets to train the model.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -96,3 +96,9 @@ tools/ffmpeg-*-static
 tools/ffmpeg-release
 tools/ffmpeg-release-*.tar.xz
 tools/Miniconda3-latest-*.sh
+
+egs2/easycom/avsr1/local/LRS3_landmarks
+egs2/easycom/avsr1/local/pre-trained
+egs2/easycom/avsr1/local/babble_noise.wav
+egs2/easycom/avsr1/local/easycom.ipynb
+egs2/easycom/avsr1/local/lrs3-valid.id
```
Comment (sw005320) [inline .gitignore:104]: These lines depend on your environment.
These should be removed.

File: egs2/TEMPLATE/asr1/db.sh [hunk 2]
```diff
@@ -111,6 +111,7 @@ COMMONVOICE=downloads
 CVSS=
 FLEURS=downloads
 MICROSOFT_SPEECH_CORPUS=
+EASYCOM=/mnt/ssd2/Dataset/EasyComDataset
```
Comment (sw005320) [inline egs2/TEMPLATE/asr1/db.sh:114]:
```suggestion
EASYCOM=
```

File: egs2/TEMPLATE/asr1/db.sh [hunk 3]
```diff
@@ -157,7 +158,8 @@ GOOGLEI18N=downloads
 NOISY_SPEECH=
 NOISY_REVERBERANT_SPEECH=
 LRS2=
-LRS3=
+LRS3=/mnt/hard/dataset/LRS3
```
Comment (sw005320) [inline egs2/TEMPLATE/asr1/db.sh:161]:
```suggestion
LRS3=
```

File: egs2/TEMPLATE/asr1/db.sh [hunk 4]
```diff
@@ -157,7 +158,8 @@ GOOGLEI18N=downloads
...
+EASYCOM=
```
Comment (sw005320) [inline egs2/TEMPLATE/asr1/db.sh:162]: duplicated

File: egs2/easycom/avsr1/README.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+|---|---|---|---|---|---|---|---|---|
+|inference_asr_model_valid.acc.ave/test_with_LRS3|694|26995|82.2|8.6|9.2|5.7|23.5|75.4|
+
+### TER
+
+|dataset|Snt|Wrd|Corr|Sub|Del|Ins|Err|S.Err|
+|---|---|---|---|---|---|---|---|---|
+|inference_asr_model_valid.acc.ave/test_with_LRS3|694|8886|70.4|18.6|11.0|5.0|34.6|75.4|
+
+## Audio-only Speech Recognition Results (Audio-only) <br> exp/asr_train_avsr_avhubert_large_with_lrs3_noise_extracted_en_bpe1000
```
Comment (sw005320) [inline egs2/easycom/avsr1/README.md:58]: The audio-only data significantly degrades the performance.
Can you provide some reasons?
Probably due to the AV HuBERT architecture?

**Review by sw005320 — COMMENTED** [review]
I added several comments.

File: egs2/easycom/avsr1/local/data.sh [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+
+. ./db.sh
+. ./path.sh
+
+cmd=run.pl
```
Comment (ftshijt) [inline egs2/easycom/avsr1/local/data.sh:16]: I guess you may simply do `. ./cmd.sh`?

File: egs2/easycom/avsr1/local/data.sh [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+    if [ -z "${LRS3}" ]; then
+        log "Fill the value of 'LRS3' of db.sh"
+        log "Dataset can be download from https://mmai.io/datasets/lip_reading/"
+        exit 1
+    fi
+fi
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    # Data preprocessing (audio segmentation & face crop & text preprocess)
+    # Session 4,12 : Validation / Session 10,11 : Test
```
Comment (ftshijt) [inline egs2/easycom/avsr1/local/data.sh:53]: could you explain a bit on this comment? seems a bit of confusion to me

**Review by ftshijt — COMMENTED** [review]
Very cool extension! Many thanks for the effort.

Could you please also add en entry in egs2/README.md for the dataset?

Also, two minor comments as follows:

File: egs2/easycom/avsr1/local/data.sh (hunk 7 above)
Comment (ftshijt) [reply egs2/easycom/avsr1/local/data.sh]: Oh, I see. thanks for the clarification, then it should be good!

**Comment by sw005320** [comment]
Thanks a lot!
